agent/agent.go

```diff
@@ -906,6 +906,14 @@ func (a *agent) handleManifest(manifestOK *checkpoint) func(ctx context.Context,
 		}
 		a.client.RewriteDERPMap(manifest.DERPMap)
 
+		seed, err := stringToInt64Hash(manifest.WorkspaceID)
+		if err != nil {
+			return xerrors.Errorf("generate hash from workspace id: %w", err)
+		}
+		if err := a.sshServer.UpdateHostSigner(seed); err != nil {
```

Comment:
It makes more sense to me to put this in `createTailnet` just before we start listening for SSH connections. The manifest will be available at that time, and the whole point is to have a stable key, so we don't need to run it more than once.